### PR TITLE
[[ SVG ]] Revision 3

### PR DESCRIPTION
**Pull request by runrevmark**
This patch contains a number of improvements / fixes to the drawing / svg support:

- The viewport defined by a drawing is now completely static. This means that a drawing always has an easily accessible and well defined width/height.

- Drawing-style images now rotate, can be rendered from LCB and work correctly as icons.

- The drawing format has been refined, simplifying opcodes and removing unnecessary ones.

### CODE REVIEW

**Comment by bwmilby** [comment]
Very nice.  Any feel for a performance difference between using SVG path data and a compiled drawing for icons within widgets?

If a compiled version of a drawing is saved to disk (.csvg or .lcdwg maybe), wouldn't `image from file` work as well?

**Comment by runrevmark** [comment]
@bwmilby : I wouldn't have thought any performance difference (if there is any) would be noticeable - the cost of rasterizing pretty much anything will dominate the overhead of actually making the calls. Drawings don't require any real processing to use - they are basically just a canned sequenced of calls on an idealized graphics context, so it should be pretty much equivalent to using an OnPaint handler which renders the same thing (if not slightly faster).

At the moment I don't believe ```image from file``` will work - using file-based images uses the engine's internal image cacheing logic, which doesn't work with 'Vector' images (instances of class MCVectorImageRep). We need to do some work there to unify them - I've stared at the code a bit, but not yet seen an appropriate way to do so!

**Comment by runrevmark** [comment]
Removing WIP label as I think this is 'good to go' :)

File: extensions/script-libraries/drawing/drawing.livecodescript [hunk 1]
```diff
@@ -616,12 +616,62 @@ end _svgCascade
  ******************************************************************************/
 
 private command _svgCompile @xContext, pRootElement
-	put pRootElement["features"]["width"] into xContext["drawing-width"]
-	put pRootElement["features"]["height"] into xContext["drawing-height"]
-	put pRootElement["features"]["viewBox"] into xContext["drawing-viewbox"]
-	put pRootElement["features"]["preserveAspectRatio"] into xContext["drawing-par"]
-	delete variable pRootElement["features"]["viewBox"]
-	delete variable pRootElement["features"]["preserveAspectRatio"]
+	/* If the root element is an 'svg' then there's not much we can do! */
```
Comment (livecodeali) [inline extensions/script-libraries/drawing/drawing.livecodescript]: isn't!

**Comment by livecodeali** [comment]
Think the only thing I can see is that comment typo!

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 4201052

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 420105255aaf73ec11c6675de80bec16ef7463bd

**Comment by livecode-vulcan** [comment]
😎 test success 420105255aaf73ec11c6675de80bec16ef7463bd

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1168)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1184)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1158)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/111)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/111)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1163)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1185)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1168)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1184)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1167)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1177)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1201)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1162)

**Pull request merged**